File: admin/src/main/java/com/jigumulmi/place/dto/request/AdminCreateTemporaryBusinessHourRequestDto.java
package com.jigumulmi.place.dto.request;

import com.jigumulmi.place.dto.BusinessHour;
import com.jigumulmi.place.dto.validator.ValidBusinessHour;
import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Getter
@NoArgsConstructor
@SuperBuilder
public class AdminCreateTemporaryBusinessHourRequestDto {

    @NotNull
    @FutureOrPresent
    private LocalDate date;
    @ValidBusinessHour
    private BusinessHour businessHour;
}

File: admin/src/main/kotlin/com/jigumulmi/place/dto/request/AdminCreateTemporaryBusinessHourRequestDto.kt
package com.jigumulmi.place.dto.request

import com.jigumulmi.place.dto.BusinessHour
import com.jigumulmi.place.dto.validator.ValidBusinessHour
import jakarta.validation.constraints.FutureOrPresent
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

data class AdminCreateTemporaryBusinessHourRequestDto (
    val date: @NotNull @FutureOrPresent LocalDate,

    @ValidBusinessHour
    val businessHour: BusinessHour,
)

